### DISCUSSION TITLE
Configuring Dap Adapters

### DISCUSSION THREAD

**Discussion by Jeordanis** [nix-community/nixvim#921] [Q&A] [answered]
Hey there! First of all, good job and thank you!

I'm new to all this but decided to give it a try. I'm trying to configure nixvim for cpp debugging. 
TBH I don't know what I'm doing. This is my current config file:

`{
  # Import all your configuration modules here
  imports = [
    ./bufferline.nix
  ];

  config = {

    colorschemes.onedark.enable = true;
    options = {
      number = true;
      relativenumber = true;
      shiftwidth = 2;
    };

    globals.mapleader = " ";

    keymaps = [
      {
        action = "<cmd>Telescope live_grep<CR>";
        key = "<leader>g";
      }
    ];
    
    plugins = {
      lualine.enable = true;
      telescope.enable = true;
      clangd-extensions.enable = true;
      dap.enable = true;
      floaterm.enable = true;
      which-key.enable = true;
    };

    plugins.dap = {
      adapters = {
        codelldb.enable = true;
      };
    };

    plugins.lsp = {
      enable = true;
      servers = {
        clangd.enable = true;
      };
    };
  };

}
`
Any help is appreciated.

**Comment by GaetanLepage** [comment]
Hello !
Thanks and welcome :)
I am not sure to understand what is wrong exactly. Is your current setup working ? Do you simply want to extend/improve it ?
Were you using neovim prior to nixvim ?

In any case, you can start by looking at two different resources:
- The [DAP documentation](https://microsoft.github.io/debug-adapter-protocol/implementors/adapters/) to be able to see which adapter might be the best suited for your use case. If it happens not to be supported by nixvim, please, feel free to open an issue on the repo.
- The [nixvim dap documentation](https://nix-community.github.io/nixvim/plugins/dap/index.html) to see what options/adapters are available.

Also, feel free to join [the matrix room](https://matrix.to/#/#nixvim:matrix.org) where other users can help you to get started and improve your config.

Happy hacking !

**Reply by Jeordanis** [reply]
Thanks for your response. 

My bad, I forgot to tell what is the problem. When I launch nixvim I get an error stating that plugins.dap.adapters.codepldb doesn’t exist.

I got the same error with any adapter that I tried, so I assumed that I was doing it wrong. Perhaps they are not implemented?

**Reply by GaetanLepage** [reply, marked as answer]
Well, we don't have the support for this adapter, but you should still be able to use it.
The actual options are `dap.adapters.executables` and `dap.adapters.servers`.
See [this file](https://github.com/nix-community/nixvim/blob/6645fea915ecf17cf465d544fd42f0f549c00c14/tests/test-sources/plugins/dap/nvim-dap.nix) for an example.

**Reply by Jeordanis** [reply]
Thanks! I will give it a try.
